### PR TITLE
Create new branch if branch doesn't exist.

### PR DESCRIPTION
**Pull request by beausmith**

### CODE REVIEW

File: bin/git-checkout-plus [hunk 1]
```diff
@@ -107,5 +107,9 @@ elif [[ "${1}" =~ '^https://github.com/([-_0-9a-zA-Z]+)/([-_0-9a-zA-Z]+)/pull/([
     fi
   fi
 else
-  git checkout "$@"
+  if git-branch-exists "$@"; then
+    git checkout "$@"
+  else
+    git checkout -b "$@"
```
Comment (beausmith) [inline bin/git-checkout-plus:113]: Perhaps a message "New branch created…" should also be output here to ensure user knows that branch did not previously exist.

Comment (eventualbuddha) [reply bin/git-checkout-plus:113]: I often just run `gco -b foo`. I think this change would be fine as long as it only does it if there's only a single argument and it's not a flag (i.e. doesn't start with `-`).

**Comment by beausmith** [comment]
Closing this feature as it interferes in the ability to automatically check out a remote branch of the same name using `git checkout`.

File: bin/git-checkout-plus (hunk 1 above)
Comment (beausmith) [reply bin/git-checkout-plus:113]: image

**Pull request closed without merging**